components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/jms/JMSInjectHandler.java

```diff
@@ -97,21 +96,26 @@ public boolean invoke(Object object, String name) throws SynapseException{
             msgCtx.setProperty("inbound.endpoint.name", name);
             InboundEndpoint inboundEndpoint = msgCtx.getConfiguration().getInboundEndpoint(name);
             CustomLogSetter.getInstance().setLogAppender(inboundEndpoint.getArtifactContainerName());
-            String contentType = msg.getJMSType();
-            
-            if (contentType == null || contentType.trim().equals("")) {
-                String contentTypeProperty =
-                                             jmsProperties.getProperty(JMSConstants.CONTENT_TYPE_PROPERTY);
-                if (contentTypeProperty != null) {
-                    contentType = msg.getStringProperty(contentTypeProperty);
-                }
-            }else{
-                msgCtx.setProperty(JMSConstants.JMS_MESSAGE_TYPE, contentType);
+            String contentType = null;
+
+            String contentTypeProperty =
+                    jmsProperties.getProperty(JMSConstants.CONTENT_TYPE_PROPERTY);
+            if (contentTypeProperty != null) {
+                contentType = msg.getStringProperty(contentTypeProperty);
             }
-            
-            if(contentType == null || contentType.trim().equals("")){
+
+            if (contentType == null || contentType.trim().equals("")) {
```

Comment:
Shall we use the isEmpty method in here?

Reply:
Fixed in b6ecdd3